### PR TITLE
fix : range data labels on box series

### PR DESCRIPTION
**Pull request by jungeun-cho**
### Please check if the PR fulfills these requirements
- [ ] It's submitted to right branch according to our branching model
- [ ] It's right issue type on title
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has description for the breaking change

### Description
**Bar**

| AS-IS | TO-BE |
| --- | --- |
| 스크린샷 2021-01-19 18 10 37 | 스크린샷 2021-01-19 18 04 48 |

**Column**
| AS-IS | TO-BE |
| --- | --- |
| 스크린샷 2021-01-19 18 11 36 | 스크린샷 2021-01-19 18 06 07 |

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: apps/chart/src/component/boxSeries.ts [hunk 1]
```diff
@@ -732,15 +738,44 @@ export default class BoxSeries extends Component {
...
+          size: this.getOffsetSize(),
+        },
+        theme: omit(this.theme.dataLabels, 'stackTotal'),
+      };
+
+      return [...acc, data];
+    }, [] as RectDataLabel[]);
+  }
+
+  getDataLabelRangeDataDirection(value: number, index: number) {
```
Comment (jung-han) [inline apps/chart/src/component/boxSeries.ts:759]: isEven을 그냥 인자로 받는것은 어떤가요?

File: apps/chart/src/component/boxSeries.ts [hunk 2]
```diff
@@ -345,7 +345,13 @@ export default class BoxSeries extends Component {
     }
 
     if (getDataLabelsOptions(options, this.name).visible) {
-      const dataLabelData = seriesModels.map((data) => this.makeDataLabel(data, centerYAxis));
+      const dataLabelData = seriesModels.reduce<RectDataLabel[]>((acc, data) => {
+        const labelData = isRangeValue(data.value)
+          ? this.makeDataLabelRangeData(data)
+          : [this.makeDataLabel(data, centerYAxis)];
```
Comment (jung-han) [inline apps/chart/src/component/boxSeries.ts:351]: `makeDataLabel`도 배열을 반환해줘서 맞춰주는것은 어떤가요?_?

File: apps/chart/src/component/boxSeries.ts [hunk 3]
```diff
@@ -732,15 +738,44 @@ export default class BoxSeries extends Component {
...
+        plot: {
+          x: 0,
+          y: 0,
+          size: this.getOffsetSize(),
+        },
+        theme: omit(this.theme.dataLabels, 'stackTotal'),
+      };
+
+      return [...acc, data];
+    }, [] as RectDataLabel[]);
```
Comment (jung-han) [inline apps/chart/src/component/boxSeries.ts:756]:
```ts
return (rect.value as RangeDataType<number>).reduce<RectDataLabel[]>(
      (acc, value, index) => [
        ...acc,
        {
          ...rect,
          value,
          direction: this.getDataLabelRangeDataDirection(value, index),
          plot: {
            x: 0,
            y: 0,
            size: this.getOffsetSize(),
          },
          theme: omit(this.theme.dataLabels, 'stackTotal'),
        },
      ],
      []
    );
```

로 줄일 수 있겠네요!

**Review by jung-han — APPROVED** [review]
완료입니다!

**Review by shiren — APPROVED** [review]
LGTM

**Review by lja1018 — APPROVED** [review]
리뷰완료
고생하셨습니다!

File: apps/chart/src/component/boxSeries.ts (hunk 2 above)
Comment (jungeun-cho) [reply apps/chart/src/component/boxSeries.ts:351]: BoxStack에서도 공통으로 사용하고 있습니다. 아래처럼 수정했습니다~
```ts
const dataLabelData = seriesModels.reduce<RectDataLabel[]>((acc, data) => {
  return isRangeValue(data.value)
    ? [...acc, ...this.makeDataLabelRangeData(data)]
    : [...acc, this.makeDataLabel(data, centerYAxis)];
  }, []);
```

**Pull request merged**